lib/Doctrine/Persistence/Mapping/Driver/AnnotationDriver.php

```diff
@@ -61,6 +61,7 @@ abstract class AnnotationDriver implements MappingDriver
      * Cache for AnnotationDriver#getAllClassNames().
      *
      * @var string[]|null
+     * @psalm-var list<class-string>|null
```

Comment:
Nit: if this is an array, we could initialise it to `[]` and avoid `null`, but I'll admit I haven't checked if the code has any special handling when it is indeed `null`. We can also make the change in 2.3 if you don't want to make this kind of change in a patch release.

Reply:
There is in fact special handling here: 

https://github.com/doctrine/persistence/blob/8b420f7065c2d6c0149ff523816f6b28391581d1/lib/Doctrine/Persistence/Mapping/Driver/AnnotationDriver.php#L194-L196

I must admit I didn't check the code either before you asked.

Reply:
Then let's not make the change here but in a separate PR 👍